### PR TITLE
[IMP] web: Add core components documentation

### PR DESCRIPTION
**Pull request by Iucapad**
This commit adds the documentation for core components that have been improved or moved during the Studio refactor. Some new props have been added, and the documentation is now present as expected.

Components involved:
- Notebook
- SelectMenu
- TagsList

### CODE REVIEW

**Comment by Iucapad** [comment]
Hello @fdardenne Don't hesitate to check and tell me if something seems wrong in the documentation. This PR is related to changes made during the studio refactor

File: content/developer/reference/frontend/owl_components.rst [hunk 1]
```diff
@@ -712,6 +714,10 @@ at the top of the element to display horizontally, or at the left for a vertical
 There are two ways to define your Notebook pages to instanciate, either by using `slot`'s,
 or by passing a dedicated `props`.
 
+A page can be made disabled with the `isDisabled` attribute, set directly on the slot node, or
```
Comment (fdardenne) [inline content/developer/reference/frontend/owl_components.rst:717]: Why not:
```suggestion
A page can be disabled with the `isDisabled` attribute, set directly on the slot node, or
```

File: content/developer/reference/frontend/owl_components.rst [hunk 2]
```diff
@@ -993,3 +1011,98 @@ The shape of a `group` is the following:
...
+
+   In the next example, a TagsList component is used to display multiple tags.
+   It's at the developer to handle from the parent what would happen when the tag is pressed, or when the delete button is clicked.
+
+   .. code-block:: javascript
+
+      import { TagsList } from "@web/core/tags_list/tags_list";
+
+      class Parent extends Component {
+        setup() {
```
Comment (fdardenne) [inline content/developer/reference/frontend/owl_components.rst:1081]: The indentation here is not properly set: there is only 2 spaces here compared to 4 spaces below

**Review by fdardenne — APPROVED** [review]
Clean 👍 
Just a small comment

File: content/developer/reference/frontend/owl_components.rst (hunk 2 above)
Comment (Iucapad) [reply content/developer/reference/frontend/owl_components.rst:1081]: You're right, I copy/pasted from a test with a messed up indentation

**Comment by Iucapad** [comment]
@fdardenne It should be fine now, thanks for the review

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**